projects/ngneat/transloco-locale/src/lib/transloco-locale.service.ts

```diff
@@ -0,0 +1,61 @@
+import {Injectable, Inject, OnDestroy} from '@angular/core';
+import { TranslocoService, HashMap } from '@ngneat/transloco';
+import {Observable, BehaviorSubject, Subscription} from 'rxjs';
+import { map, distinctUntilChanged, filter } from 'rxjs/operators';
+import { isLocaleFormat } from './helpers';
+import { LOCALE_LANG_MAPPING, LOCALE_DEFAULT_LOCALE } from './transloco-locale.config';
+import { Locale } from './transloco-locale.types';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TranslocoLocaleService implements OnDestroy{
+  localeChanges$: Observable<Locale>;
+  private locale: BehaviorSubject<Locale>;
+  private _locale: Locale | null;
+  private subscription: Subscription;
+
+  constructor(
+    private translocoService: TranslocoService,
+    @Inject(LOCALE_LANG_MAPPING) private langLocaleMapping: HashMap<Locale>,
+    @Inject(LOCALE_DEFAULT_LOCALE) private defaultLocale: Locale
+  ) {
+    this._locale = defaultLocale || this.toLocale(this.translocoService.getActiveLang());
+    this.locale = new BehaviorSubject(this._locale);
+    this.localeChanges$ = this.locale.asObservable().pipe(distinctUntilChanged());
+
+    this.subscription = translocoService.langChanges$
+      .pipe(
+        map(this.toLocale.bind(this)),
+        filter(lang => !!lang)
+      )
+      .subscribe(this.setLocale.bind(this));
+  }
+
+  getLocale() {
+    return this._locale;
+  }
+
+  setLocale(locale: Locale) {
+    if (!isLocaleFormat(locale)) {
+      console.error(`${locale} isn't a valid locale format`);
```

Comment:
What happens when given locale is not in proper format? Does it not break anything? Because, if it does, then this condition should probably throw an error instead of logging it to the console.

And I would recommend using `isDevMode` before logging anything to the console. https://angular.io/api/core/isDevMode